### PR TITLE
feat: switching over to new docsearch

### PR DESCRIPTION
**Pull request by russorat**
we are moving over to the algolia search where they crawl and index our site from their end. should improve docs search like 1000000%

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -39,18 +39,4 @@ jobs:
...
-      - name: Run algolia/docsearch-scraper
-        run: |
-          docker pull algolia/docsearch-scraper
-          docker run \
-            -e ALGOLIA_APP_ID=${{ secrets.DEV_ALGOLIA_APPLICATION_ID }} \
-            -e API_KEY=${{ secrets.DEV_ALGOLIA_API_KEY }} \
-            -e INDEX_NAME_TMP="prod_ngrok_tmp_$(date +%s)"  \
-            -e "CONFIG=$(cat ./algolia_prod.config.json | jq -r tostring)" \
-            algolia/docsearch-scraper
+          aws s3 sync ./build s3://docs-s3.ngrok.com/docs --acl public-read
```
Comment (krwenholz) [inline .github/workflows/deploy.yml:42]: This shouldn't show as a newline. I think the tabbing is off?

**Review by krwenholz — CHANGES_REQUESTED** [review]
I think indentation is off for the workflows. Otherwise looks correct!

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (russorat) [reply .github/workflows/deploy.yml:42]: that's strange, the tabbing looks all good on my side. not sure how to fix

Comment (krwenholz) [reply .github/workflows/deploy.yml:42]: Weird. If it looks okay locally and no mixed tabs/spaces, that's okay!

**Review by krwenholz — APPROVED** [review]
We chatted. Mine and GitHub's eyes must just be bad.

**Pull request merged**